block/manager.go

```diff
@@ -236,3 +242,43 @@ func (m *Manager) MustLoadCommit(h uint64) *types.Commit {
 	}
 	return ret
 }
+
+func (m *Manager) AddBlockToCache(h uint64, b *types.Block, c *types.Commit) {
+	m.retrieverMu.Lock()
```

Comment:
I think better to change the lock to `blockCacheMu` as it's only being used for block cache.
I think one of the pr does it already.
Also I may move all of its to a seperate file called blockCache.